e2e-pw/src/oss/poms/color-modal/index.ts

```diff
@@ -0,0 +1,127 @@
+  readonly page: Page;
+  readonly colorModal: Locator;
+
+  constructor(page: Page) {
+    this.page = page;
+    this.colorModal = page.locator("#colorModal");
+  }
+
+  async closeColorModal() {
+    const closeButton = this.colorModal.getByTestId("close-color-modal");
+    return closeButton.click();
+  }
+
+  async selectActiveField(fieldName: string) {
+    const fieldSelector = this.page.getByTestId(
+      `color-modal-list-item-${fieldName}`
+    );
+    return fieldSelector.click();
+  }
+
+  async setColorBy(mode: "value" | "field") {
+    const globalTab = this.page.getByTestId(
+      "color-modal-list-item-Global settings"
+    );
+    await globalTab.click();
+    const modeButton = this.page.getByTestId(`radio-button-${mode}`);
+    return modeButton.click();
+  }
+
+  async shuffleColors() {
+    const globalTab = this.page.getByTestId(
+      "color-modal-list-item-Global settings"
+    );
+    await globalTab.click();
+    const shuffleButton = this.page.getByTestId("shuffle-colors");
+    return shuffleButton.click();
+  }
+
+  async useColorBlindColors() {
+    const globalTab = this.page.getByTestId(
+      "color-modal-list-item-Global settings"
+    );
+    await globalTab.click();
+    const colorSettingCheckbox = this.page.getByTestId(
+      "checkbox-Use color blind friendly option"
+    );
+    return colorSettingCheckbox.click();
+  }
+
+  async setOpacity(value: number) {}
+
+  async useMultiColorKeypoints() {
+    const globalTab = this.page.getByTestId(
+      "color-modal-list-item-Global settings"
+    );
+    await globalTab.click();
+    const useMultiColorKeypointsCheckbox = this.page.getByTestId(
+      "checkbox-Multicolor keypoints"
+    );
+    return useMultiColorKeypointsCheckbox.click();
+  }
+
+  // field level setting
+  async toggleColorMode() {
+    const toggleButton = this.page.getByTestId(
+      "button-toggle between color by value or color by field mode"
+    );
+    return toggleButton.click();
+  }
+
+  async useSpecialFieldColor(fieldName: string) {
+    return this.page
+      .getByTitle(`Use custom color for ${fieldName} field`)
+      .first()
+      .click({ force: true });
+  }
+
+  async setSpecialFieldColor(color: string) {
+    const container = this.page.getByTestId("field-color-div");
+    await container.isVisible();
+    await container.getByRole("textbox").fill(color);
+  }
+
+  // value level setting
+  async selectColorByAttribute(field: string) {
+    await this.colorModal
+      .getByTestId("select-attribute")
+      .click({ force: true });
+    await this.colorModal
+      .getByTestId(`filter-option-${field}`)
+      .click({ force: true });
+  }
+
+  async addANewPair(value: string, color: string) {
+    const addButton = this.page.getByTestId("button-add a new pair");
+    await addButton.click();
+    await this.page.getByPlaceholder("Value (e.g. 'car')").last().fill(value);
+    await this.page.getByPlaceholder("#dd00dd").last().fill(color);
```

Comment:
cool! I don't know how but it might be valuable for us to somehow use theme in the tests too for color/positioning/spacing assertions. 🤔 just thoughts